Fosshub - Changing to use core extension

gep13 opened the pull request.

gep13 commented:
@majkinetor this is the changes I was talking to you about.

Trying to change the Fosshub extension to use the new Get-WebContent helper method that was added to the chocolatey-core.extension.

However, when I try to run `.\update.ps1` in the audacity folder locally having made these changes, I get the following:

![image](https://cloud.githubusercontent.com/assets/1271146/19627425/7390503c-993e-11e6-9e84-faca5791d1f6.png)

So I changed the update.ps1 to include this:

```
import-module .\..\..\extensions\chocolatey-core.extension\extensions\Get-WebContent.psm1
```

But this then resulted in the following:

![image](https://cloud.githubusercontent.com/assets/1271146/19627433/92deca68-993e-11e6-975e-59e5be3d956b.png)

If you have a minute, can you look into what might be going on here?

Thanks!

majkinetor commented:
I pushed a new version of AU that fixes this problem. I got new errors tho, but those are not related to GetWeb-Content:

```
Set-Content : Could not find a part of the path 'C:\Users\majkinetor\AppData\Local\Temp\chocolatey\audacity\2.1.2.20161023\ro3ojwdm.1kb.istext'.
At C:\Users\majkinetor\AppData\Local\Temp\chocolatey\au\chocolatey\helpers\functions\Get-WebFile.ps1:205 char:13
+             Set-Content -Path "$fileName.istext" -Value "$fileName ha ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\MAJKIN...jwdm.1kb.istext:String) [Set-Content], DirectoryNotFoundException
    + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.PowerShell.Commands.SetContentCommand


Download of ro3ojwdm.1kb (1,24 KB) completed.
Attempt to use original download file name failed for 'Audacity.html'.
Copying audacity
  from 'Audacity.html'
Copy-Item : Cannot find path 'C:\work\_chocolatey\chocolatey-coreteampackages\automatic\audacity\Audacity.html' because it does not exist.
At C:\Users\majkinetor\AppData\Local\Temp\chocolatey\au\chocolatey\helpers\functions\Get-ChocolateyWebFile.ps1:350 char:5
+     Copy-Item $url -Destination $fileFullPath -Force

gep13 commented:
@majkinetor when you say you pushed a new version, what does that mean?  Can I update through PowerShell Gallery?

majkinetor commented:
No, download latest artifact, unpack it and run install.ps1.
Alternativelly, you can git clone au locally and `./build -Install`.

gep13 commented:
@majkinetor okay, installed locally, after updating my clone.  Thanks!

Any thoughts on this new error then?  Is that a package issue? Or something else?

majkinetor commented:
Didn't have time to investigate it yet.

gep13 commented:
Okay, not a problem.

majkinetor commented on extensions/chocolatey-fosshub.extension/extensions/Get-UrlFromFosshub.psm1:
```diff
@@ -32,10 +32,8 @@ Function Get-UrlFromFosshub($linkUrl) {
-
-  $htmlPage = $webClient.DownloadString($linkUrl)
+  $opts = @{ Headers = @{ Referer = 'https://www.fosshub.com/${fosshubAppName}' } }
+  $htmlPage = Get-WebContent -url "http://example.com" -options $opts
```
Yeah, no wonder it doesn't work

majkinetor commented:
```
10 [ M:1 D:0 A:0 ?:1 ]  16-10-29 17.37.31  C:\work\_chocolatey\chocolatey-coreteampackages\automatic\qbittorrent
FosshubChanges~> .\update.ps1
qbittorrent - checking updates using au version 2016.10.29
nuspec version: 3.3.7
remote version: 3.3.7
No new version found, but update is forced
New version is available
Automatic checksum started
WARNING: C:\Users\MAJKIN~1\AppData\Local\Temp\chocolatey\qbittorrent\3.3.7.20161029\4hf3nxgy.a2d is of content type text/html; charset=utf-8

Download of 4hf3nxgy.a2d (-1 B) completed.
Downloading qBittorrent 32 bit
  from 'https://download.fosshub.com/Protected/expiretime=1477755537;badurl=aHR0cDovL3d3dy5mb3NzaHViLmNvbS9xQml0dG9ycmVudC5odG1s/343790ce2028d24e7d99b8dbc3a4a8c405a2ceeae49c7f0c9b5fb390bc432c05/qBittorrent/qbittorrent_3.3.7_setup.exe'

Download of qbittorrent_3.3.7_setup.exe (15,7 MB) completed.
Package downloaded and hash calculated for 32 bit version
Updating files
  $Latest data:
    Checksum32      (String)    49ae9a0adfc3272bec38822c528f732d9495b79a2a7ca934f8c6635237b15d07
    ChecksumType32  (String)    sha256
    NuspecVersion   (String)    3.3.7
    PackageName     (String)    qbittorrent
    URL32           (String)    https://www.fosshub.com/qBittorrent.html/qbittorrent_3.3.7_setup.exe
    Version         (String)    3.3.7.20161029

  qbittorrent.nuspec
    setting id:  qbittorrent
    using Chocolatey fix notation: 3.3.7 -> 3.3.7.20161029
  .\tools\chocolateyInstall.ps1
    (^[$]checksumType\s*=\s*)('.*') = $1'sha256'
    (^[$]fosshubUrl\s*=\s*)('.*') = $1'https://www.fosshub.com/qBittorrent.html/qbittorrent_3.3.7_setup.exe'
    (^[$]checksum\s*=\s*)('.*') = $1'49ae9a0adfc3272bec38822c528f732d9495b79a2a7ca934f8c6635237b15d07'
Attempting to build package from 'qbittorrent.nuspec'.
Successfully created package 'C:\work\_chocolatey\chocolatey-coreteampackages\automatic\qbittorrent\qbittorrent.3.3.7.20161029.nupkg'
Package updated
```

gep13 commented:
@majkinetor  Doh! Thanks for taking a look at this, and fixing it up.

I have just rebased this PR to bring it up to date with the latest master branch, and I was confused to find that the commit where I moved the packages to use the new core extension was removed.  When I checked, it looks like it already got merged into master.

This happened here:

![image](https://cloud.githubusercontent.com/assets/1271146/19835606/1baa9c60-9e84-11e6-91a6-7dd10a9c9158.png)

Not entirely sure how that managed to happen, but it looks like it got pulled in when you were merging the jitsi package.

All that was left in this PR was the single commit that is still here, where you changed example.com to the correct `$linkUrl` variable, and another commit from you about removing accidential commit.  This commit was deleting the jitsi update.ps1 file, which I don't think we want to happen, so I removed that commit from this PR.

gep13 commented:
@majkinetor also, I still don't get the above `update.ps1` to work locally, I still see the `Get-WebContent` error.  Have you done anything else different on your machine to get it to work?

I have updated to the latest AU version, thinking that you added something in there, but still no joy.

majkinetor commented:
I was moving between branches and when quickly testing your stuff back then I accidentally pushed some of my unreleated work - so its probably merged because of it.... sorry.  I took care that it doesn't happen in the future, current branch is now VERY visible in shell.

I will recheck again to see if I can reproduce.

gep13 commented:
>@majkinetor said...
current branch is now VERY visible in shell.

What changes have you made?

>@majkinetor said...
I will recheck again to see if I can reproduce.

Thank you, appreciate it!

majkinetor commented:
> What changes have you made?

SImilar to [posh-git] but in my prompt function within my profile.d. (https://github.com/majkinetor/powershell_profile.d/blob/master/10_prompt.ps1#L13-L29).

majkinetor commented:
Fixed in au.
